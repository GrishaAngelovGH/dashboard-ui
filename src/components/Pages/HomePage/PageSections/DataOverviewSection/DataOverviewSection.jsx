import { Fade } from 'react-awesome-reveal'
import PageSection from 'components/Pages/PageSection'

import { dataTypes } from 'images'
import './DataOverviewSection.scss'

const DataOverviewSection = ({ types }) => (
	<Fade duration={3000}>
		<PageSection style={{ background: '#2a7ab9', height: 520 }}>
			<div className='row no-gutters'>
				<div className='col-md-12 mt-2'>
					<div className='row no-gutters justify-content-center m-3'>
						<div className='col-md-8 text-center'>
							<h2 className='text-white'>Work with data</h2>
						</div>
					</div>

					<div className='row no-gutters justify-content-center mb-5 text-white text-center'>
						<div className='col-md-8 col-11'>
							The dashboards components supports more than 15 types
							of data servers - MS SQL, Oracle, PostgreSQL, Firebird, and others.
							Our products support all standard types of ADO.NET objects,
							the OData protocol, the ability to load data from XML, JSON,
							Excel and CSV files, work with multi-level business objects.
						</div>
					</div>

					<div className='row no-gutters justify-content-center mb-5'>
						<div className='col-md-7 col-12 d-none d-md-block type-images'>
							{
								types.map(v => (
									<img
										key={v}
										width={50}
										height={50}
										src={dataTypes[v]}
										className='m-2'
									/>
								))
							}
						</div>
					</div>

					<div className='row no-gutters justify-content-center mb-2 text-white text-center'>
						<div className='col-md-8 col-11'>
							Preliminary data preparation and conversion are available.
							You may use virtual data sources, and you may embed data directly
							into the template in the form of resources.
							Do not think about how to prepare data for the report - Dashboards will do it for you!
						</div>
					</div>
				</div>
			</div>
		</PageSection>
	</Fade>
)

export default DataOverviewSection