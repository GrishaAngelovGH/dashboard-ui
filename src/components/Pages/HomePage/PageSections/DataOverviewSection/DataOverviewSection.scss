@media screen and (max-width: 768px) {
	.type-images > img {
		width: 20px;
		height: 20px;
	}
}
