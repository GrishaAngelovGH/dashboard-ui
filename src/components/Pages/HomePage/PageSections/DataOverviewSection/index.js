import DataOverviewSection from './DataOverviewSection'

export default DataOverviewSection