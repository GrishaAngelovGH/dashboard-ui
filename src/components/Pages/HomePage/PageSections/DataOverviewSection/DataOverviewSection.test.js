import { render } from '@testing-library/react'

import DataOverviewSection from './DataOverviewSection'

test('should render DataOverviewSection component', () => {
    const view = render(<DataOverviewSection types={['azure', 'csv']} />)

    expect(view).toMatchSnapshot()
})