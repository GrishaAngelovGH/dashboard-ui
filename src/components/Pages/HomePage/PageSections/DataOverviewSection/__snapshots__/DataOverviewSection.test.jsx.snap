// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render DataOverviewSection component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row no-gutters"
        style="background: rgb(42, 122, 185); height: 520px;"
      >
        <div
          class="col-md-12"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-md-12 mt-2"
            >
              <div
                class="row no-gutters justify-content-center m-3"
              >
                <div
                  class="col-md-8 text-center"
                >
                  <h2
                    class="text-white"
                  >
                    Work with data
                  </h2>
                </div>
              </div>
              <div
                class="row no-gutters justify-content-center mb-5 text-white text-center"
              >
                <div
                  class="col-md-8 col-11"
                >
                  The dashboards components supports more than 15 types of data servers - MS SQL, Oracle, PostgreSQL, Firebird, and others. Our products support all standard types of ADO.NET objects, the OData protocol, the ability to load data from XML, JSON, Excel and CSV files, work with multi-level business objects.
                </div>
              </div>
              <div
                class="row no-gutters justify-content-center mb-5"
              >
                <div
                  class="col-md-7 col-12 d-none d-md-block type-images"
                >
                  <img
                    class="wow fadeInUp m-2"
                    data-wow-delay="0.5s"
                    data-wow-duration="5s"
                    height="50"
                    src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20viewBox='0%200%2068%2068'%20enable-background='new%200%200%2068%2068'%3e%3cg%20fill='%23fff'%3e%3cpath%20d='m41%2021h6v6h-6z'/%3e%3cpath%20d='m33%2021h6v6h-6z'/%3e%3cpath%20d='m25%2021h6v6h-6z'/%3e%3cpath%20d='m25%2029h6v6h-6z'/%3e%3cpath%20d='m33%2029h6v6h-6z'/%3e%3cpath%20d='m41%2029h6v6h-6z'/%3e%3cpath%20d='m25%2037h6v6h-6z'/%3e%3cpath%20d='m33%2037h6v6h-6z'/%3e%3cpath%20d='m41%2037h6v6h-6z'/%3e%3cpath%20d='m23%2021h-4v28h3%201%2024v-4h-24z'/%3e%3cpath%20d='M49,4H19L0,34l19,30h30l19-30L49,4z%20M47,60H21L4,34L21,8h26l17,26L47,60z'/%3e%3c/g%3e%3c/svg%3e"
                    width="50"
                  />
                  <img
                    class="wow fadeInUp m-2"
                    data-wow-delay="0.5s"
                    data-wow-duration="5s"
                    height="50"
                    src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20viewBox='0%200%2068%2068'%20enable-background='new%200%200%2068%2068'%3e%3cpath%20d='m52%200h-33v21h-19v34h19v13h49v-52l-16-16m1%206l9%209h-9v-9m-26.4%2029.7c.1.2.4.5.7.7.3.2%201.1.6%202.3%201.2%201.6.8%202.7%201.6%203.3%202.4.6.8.9%201.8.9%202.9%200%201.6-.6%202.8-1.7%203.7-1.1.9-2.6%201.4-4.6%201.4-1.8%200-3.4-.4-4.9-1.1v-3.4c1.2.5%202.1.9%202.9%201.1.8.2%201.5.3%202.2.3.8%200%201.4-.2%201.8-.5.4-.3.6-.8.6-1.4%200-.3-.1-.6-.3-.9-.2-.3-.5-.5-.8-.8-.4-.2-1.1-.6-2.2-1.2-1-.5-1.8-1-2.4-1.4-.5-.5-.9-1-1.3-1.6-.3-.6-.5-1.3-.5-2.2%200-1.5.5-2.8%201.5-3.7%201-.9%202.4-1.3%204.3-1.3.9%200%201.7.1%202.5.3.8.2%201.7.5%202.5.9l-1.2%202.9c-.9-.4-1.7-.7-2.3-.8-.6-.2-1.2-.2-1.8-.2-.7%200-1.2.2-1.6.5-.4.3-.6.8-.6%201.3.5.4.6.7.7.9m-13.6%209.2c1.2%200%202.7-.3%204.4-.9v3.1c-1.4.6-3%20.9-4.7.9-2.5%200-4.4-.8-5.7-2.3-1.3-1.5-2-3.8-2-6.7%200-1.8.3-3.4%201-4.8.6-1.4%201.6-2.4%202.8-3.1%201.2-.7%202.6-1.1%204.2-1.1%201.7%200%203.3.4%205%201.2l-1.2%203c-.6-.3-1.3-.6-1.9-.8-.6-.2-1.3-.3-1.9-.3-1.4%200-2.4.5-3.2%201.6-.7%201-1.1%202.5-1.1%204.4%200%203.9%201.5%205.8%204.3%205.8m10%2019.1v-9h34v-34h-34v-17h26v15h15v45h-41m14.4-33.7h3.7l3.2%2010.4c.2.6.4%201.3.6%202.1.2.8.3%201.4.4%201.7.1-.7.4-2%20.9-3.9l3.2-10.4h3.6l-5.8%2017.5h-4l-5.8-17.4'%20fill='%23fff'/%3e%3c/svg%3e"
                    width="50"
                  />
                </div>
              </div>
              <div
                class="row no-gutters justify-content-center mb-2 text-white text-center"
              >
                <div
                  class="col-md-8 col-11"
                >
                  Preliminary data preparation and conversion are available. You may use virtual data sources, and you may embed data directly into the template in the form of resources. Do not think about how to prepare data for the report - Dashboards will do it for you!
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row no-gutters"
      style="background: rgb(42, 122, 185); height: 520px;"
    >
      <div
        class="col-md-12"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-md-12 mt-2"
          >
            <div
              class="row no-gutters justify-content-center m-3"
            >
              <div
                class="col-md-8 text-center"
              >
                <h2
                  class="text-white"
                >
                  Work with data
                </h2>
              </div>
            </div>
            <div
              class="row no-gutters justify-content-center mb-5 text-white text-center"
            >
              <div
                class="col-md-8 col-11"
              >
                The dashboards components supports more than 15 types of data servers - MS SQL, Oracle, PostgreSQL, Firebird, and others. Our products support all standard types of ADO.NET objects, the OData protocol, the ability to load data from XML, JSON, Excel and CSV files, work with multi-level business objects.
              </div>
            </div>
            <div
              class="row no-gutters justify-content-center mb-5"
            >
              <div
                class="col-md-7 col-12 d-none d-md-block type-images"
              >
                <img
                  class="wow fadeInUp m-2"
                  data-wow-delay="0.5s"
                  data-wow-duration="5s"
                  height="50"
                  src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20viewBox='0%200%2068%2068'%20enable-background='new%200%200%2068%2068'%3e%3cg%20fill='%23fff'%3e%3cpath%20d='m41%2021h6v6h-6z'/%3e%3cpath%20d='m33%2021h6v6h-6z'/%3e%3cpath%20d='m25%2021h6v6h-6z'/%3e%3cpath%20d='m25%2029h6v6h-6z'/%3e%3cpath%20d='m33%2029h6v6h-6z'/%3e%3cpath%20d='m41%2029h6v6h-6z'/%3e%3cpath%20d='m25%2037h6v6h-6z'/%3e%3cpath%20d='m33%2037h6v6h-6z'/%3e%3cpath%20d='m41%2037h6v6h-6z'/%3e%3cpath%20d='m23%2021h-4v28h3%201%2024v-4h-24z'/%3e%3cpath%20d='M49,4H19L0,34l19,30h30l19-30L49,4z%20M47,60H21L4,34L21,8h26l17,26L47,60z'/%3e%3c/g%3e%3c/svg%3e"
                  width="50"
                />
                <img
                  class="wow fadeInUp m-2"
                  data-wow-delay="0.5s"
                  data-wow-duration="5s"
                  height="50"
                  src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20viewBox='0%200%2068%2068'%20enable-background='new%200%200%2068%2068'%3e%3cpath%20d='m52%200h-33v21h-19v34h19v13h49v-52l-16-16m1%206l9%209h-9v-9m-26.4%2029.7c.1.2.4.5.7.7.3.2%201.1.6%202.3%201.2%201.6.8%202.7%201.6%203.3%202.4.6.8.9%201.8.9%202.9%200%201.6-.6%202.8-1.7%203.7-1.1.9-2.6%201.4-4.6%201.4-1.8%200-3.4-.4-4.9-1.1v-3.4c1.2.5%202.1.9%202.9%201.1.8.2%201.5.3%202.2.3.8%200%201.4-.2%201.8-.5.4-.3.6-.8.6-1.4%200-.3-.1-.6-.3-.9-.2-.3-.5-.5-.8-.8-.4-.2-1.1-.6-2.2-1.2-1-.5-1.8-1-2.4-1.4-.5-.5-.9-1-1.3-1.6-.3-.6-.5-1.3-.5-2.2%200-1.5.5-2.8%201.5-3.7%201-.9%202.4-1.3%204.3-1.3.9%200%201.7.1%202.5.3.8.2%201.7.5%202.5.9l-1.2%202.9c-.9-.4-1.7-.7-2.3-.8-.6-.2-1.2-.2-1.8-.2-.7%200-1.2.2-1.6.5-.4.3-.6.8-.6%201.3.5.4.6.7.7.9m-13.6%209.2c1.2%200%202.7-.3%204.4-.9v3.1c-1.4.6-3%20.9-4.7.9-2.5%200-4.4-.8-5.7-2.3-1.3-1.5-2-3.8-2-6.7%200-1.8.3-3.4%201-4.8.6-1.4%201.6-2.4%202.8-3.1%201.2-.7%202.6-1.1%204.2-1.1%201.7%200%203.3.4%205%201.2l-1.2%203c-.6-.3-1.3-.6-1.9-.8-.6-.2-1.3-.3-1.9-.3-1.4%200-2.4.5-3.2%201.6-.7%201-1.1%202.5-1.1%204.4%200%203.9%201.5%205.8%204.3%205.8m10%2019.1v-9h34v-34h-34v-17h26v15h15v45h-41m14.4-33.7h3.7l3.2%2010.4c.2.6.4%201.3.6%202.1.2.8.3%201.4.4%201.7.1-.7.4-2%20.9-3.9l3.2-10.4h3.6l-5.8%2017.5h-4l-5.8-17.4'%20fill='%23fff'/%3e%3c/svg%3e"
                  width="50"
                />
              </div>
            </div>
            <div
              class="row no-gutters justify-content-center mb-2 text-white text-center"
            >
              <div
                class="col-md-8 col-11"
              >
                Preliminary data preparation and conversion are available. You may use virtual data sources, and you may embed data directly into the template in the form of resources. Do not think about how to prepare data for the report - Dashboards will do it for you!
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
